### PR TITLE
Gatsby v2 for the website

### PR DESCRIPTION
**Pull request by pksjce**
Following this https://next.gatsbyjs.org/docs/migrating-from-v1-to-v2/

- Updated all the dependencies to `next` and no more dep on `gatsby-link`
- Moved `layouts` to `components` folder and added `StaticQuery`
- Added `<Layout>` component to all the pages and templates.

### CODE REVIEW

**Comment by alexreardon** [comment]
Awesome @pksjce! Keen to get this in

**Comment by pksjce** [comment]
Any idea why the `validate` is failing? The command passes on my local.

**Comment by alexreardon** [comment]
Check the build log

File: website/package.json [hunk 1]
```diff
@@ -7,12 +7,11 @@
...
-    "gatsby-link": "^1.6.46",
-    "gatsby-plugin-react-helmet": "^2.0.11",
-    "gatsby-plugin-styled-components": "^2.0.11",
-    "gatsby-source-filesystem": "^1.5.39",
-    "gatsby-transformer-remark": "^1.7.44",
+    "gatsby": "next",
+    "gatsby-plugin-react-helmet": "next",
+    "gatsby-plugin-styled-components": "next",
+    "gatsby-source-filesystem": "^next",
+    "gatsby-transformer-remark": "^next",
```
[the remark this answers is not in the document]
Comment (alexreardon) [reply website/package.json]: I am keen for it to be on version 2

[the remark this answers is not in the document]
Comment (pksjce) [reply website/package.json]: You mean I should put a version in there?

**Comment by alexreardon** [comment]
We are having build issues with this one @MarshallOfSound. It looks like the globals are not being found. Could you take a look?

**Comment by alexreardon** [comment]
I have updated the destination for this PR

**Comment by pksjce** [comment]
This became really confusing and hard to merge! So made a new PR.

**Pull request closed without merging**